exo/inference/tinygrad/models/llama.py

```diff
@@ -201,32 +206,46 @@ def __init__(
     self.freqs_cis = precompute_freqs_cis(dim // n_heads, self.max_context*2, rope_theta, rope_scaling=rope_scaling).contiguous()
     self.forward_jit = TinyJit(self.forward) if jit else None
     self.shard = shard
+    self.caches = OrderedDict()
```

Comment:
Is this a hack? I didn't think the caches would be in the model implementation itself?